pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentStateResource.java

```diff
@@ -0,0 +1,144 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.api.resources;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import org.apache.helix.AccessOption;
+import org.apache.helix.model.IdealState;
+import org.apache.pinot.common.assignment.InstancePartitions;
+import org.apache.pinot.common.assignment.InstancePartitionsUtils;
+import org.apache.pinot.common.metadata.ZKMetadataProvider;
+import org.apache.pinot.common.metadata.segment.SegmentPartitionMetadata;
+import org.apache.pinot.common.metadata.segment.SegmentZKMetadata;
+import org.apache.pinot.common.utils.LLCSegmentName;
+import org.apache.pinot.common.utils.helix.HelixHelper;
+import org.apache.pinot.controller.helix.core.PinotHelixResourceManager;
+import org.apache.pinot.controller.helix.core.assignment.segment.SegmentAssignment;
+import org.apache.pinot.controller.helix.core.assignment.segment.SegmentAssignmentFactory;
+import org.apache.pinot.controller.helix.core.assignment.segment.SegmentAssignmentUtils;
+import org.apache.pinot.controller.helix.core.realtime.PinotLLCRealtimeSegmentManager;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.apache.pinot.spi.config.table.assignment.InstancePartitionsType;
+import org.apache.pinot.spi.utils.CommonConstants;
+import org.apache.pinot.spi.utils.builder.TableNameBuilder;
+
+
+@Api(tags = Constants.SEGMENT_TAG)
+@Path("/")
+public class PinotSegmentStateResource {
+
+  @Inject
+  PinotHelixResourceManager _pinotHelixResourceManager;
+
+  // TODO add support for choosing default offset for other partitions
+  private static class RealtimeSegmentResumeConfig {
+    private Map<Integer, Long> _partitionIdToOffsetMap;
+
+    public Map<Integer, Long> getPartitionIdToOffsetMap() {
+      return _partitionIdToOffsetMap;
+    }
+
+    public void setPartitionIdToOffsetMap(Map<Integer, Long> partitionIdToOffsetMap) {
+      _partitionIdToOffsetMap = partitionIdToOffsetMap;
+    }
+  }
+
+  @POST
+  @Path("/segmentState/{tableName}/resumeRealtimeTable")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Resume a realtime table", notes = "Resume a segment")
+  public String resumeRealtimeTable(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      RealtimeSegmentResumeConfig realtimeSegmentResumeConfig) {
+    TableConfig tableConfig = _pinotHelixResourceManager.getTableConfig(tableName, TableType.REALTIME);
```

Comment:
Let's move the actual logic into the `PinotLLCRealtimeSegmentManager` and inject that into this API class.

Reply:
Ack